src/v/cluster/metadata_cache.cc

```diff
@@ -48,6 +48,16 @@ void fill_partition_leaders(
     }
 }
 
+std::optional<model::topic>
+metadata_cache::get_materialized_status(model::topic_namespace_view tp) const {
```

Comment:
maybe would be better to name it `get_source_topic` now it is confusing that it says return `status` but returns `topic`